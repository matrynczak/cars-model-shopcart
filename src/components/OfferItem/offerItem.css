#item-image {
    width: 300px;
}

.item-container {
    display: flex;
    justify-content: center;
    flex-direction: column;
    align-items: center;
    background-color: #325939;
    padding: 20px 0 10px 0;
    border-radius: 30px;
    width: 350px;
    height: 350px;
}

#item-title {
    margin-bottom: 1vh;
}

#item-attr {
    display: flex;
    flex-direction: column;
    align-items: center;
    margin: 0;
}