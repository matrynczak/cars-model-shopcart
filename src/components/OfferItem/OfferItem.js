import './offerItem.css';

function OfferItem({image, name, engine, fuel, price, id}) {
  return (
        <div className="item-container" id={id}>
            <img id="item-image" src={require(`../../assets/${image}`)}/>
                <div className='item-description'>
                <h3 id="item-title">{name}</h3>
                <p id="item-attr">
                    <span id="engine">▪ {engine}</span>
                    <span id="fuel">▪ {fuel}</span>
                    <span id="price">▪ {price} Euro</span>
                </p>
            </div>
        </div>
  );
}

export default OfferItem;
