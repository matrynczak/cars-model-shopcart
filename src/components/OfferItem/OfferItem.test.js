import renderer from 'react-test-renderer';
import OfferItem from './OfferItem';

test('Offer Item renders correctly', () => {
  const tree = renderer
  .create(<OfferItem image='./tipo-cross.jpeg' name='test' engine='test-engine' fuel='test-fuel' price='test-price' id='1'></OfferItem>)
    .toJSON();
  expect(tree).toMatchSnapshot();
});