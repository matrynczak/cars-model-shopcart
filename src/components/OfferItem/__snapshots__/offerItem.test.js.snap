// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Offer Item renders correctly 1`] = `
<div
  className="item-container"
  id="1"
>
  <img
    id="item-image"
    src="tipo-cross.jpeg"
  />
  <div
    className="item-description"
  >
    <h3
      id="item-title"
    >
      test
    </h3>
    <p
      id="item-attr"
    >
      <span
        id="engine"
      >
        ▪ 
        test-engine
      </span>
      <span
        id="fuel"
      >
        ▪ 
        test-fuel
      </span>
      <span
        id="price"
      >
        ▪ 
        test-price
         Euro
      </span>
    </p>
  </div>
</div>
`;
